### PR TITLE
Use a shorter check for a Rails app on deploy

### LINKED ISSUE #61: `deploy` in 0.9.0 sometimes fails if migrations need to be run
From @bernerdschaefer:

Environment bins expect all commands to return an exit status, but `deploy` does not always return an exit status, depending on the current state of migrations on the remote server.

### PR DESCRIPTION
**Pull request by geoffharcourt**
In 17448e9 we added a check to determine if the application was running
Rails (specifically whether the app had access to Rake and a
`db:migrate` task) through Rake's "dry-run" functionality. This approach
produces some noisy output and adds a non-trivial amount of time to the
deploy process.

Rather than start Rake and execute the dry run on every deploy, we'll
instead check for a `Rakefile` and the presence of `db/migrate` in the
application's folder structure, which will take less time and produce
less noise in `STDOUT`.

### CODE REVIEW

File: lib/parity/environment.rb [hunk 1]
```diff
@@ -132,7 +132,7 @@ def run_migration?
     end
 
     def rails_app?
-      Kernel.system("command -v rake && rake -n db:migrate")
+      Kernel.system('[ -d "db/migrate" ] && [ -f "Rakefile" ]')
```
Comment (bernerdschaefer) [inline lib/parity/environment.rb]: What about `File.directory?("db/migrate") && File.file?("Rakefile")`?

I'm mostly thinking of portability -- what shell will `system()` pick?

Comment (geoffharcourt) [inline lib/parity/environment.rb]: I like it. It also gets us running more commands in Ruby where we can capture individual command results.

Comment (geoffharcourt) [inline lib/parity/environment.rb]: Question. Is the existence of a `Rakefile` and a `db/migrate` directory good enough evidence that this is a Rails app?

Comment (bernerdschaefer) [inline lib/parity/environment.rb]: I think a `Rakefile` and a `db/migrate` directory is enough evidence to justify running `rake db:migrate`.

**Comment by croaky** [comment]
Makes sense to me.

**Pull request merged**